### PR TITLE
Add SDK documentation for `Microsoft.Testing.Platform`

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/unit-testing-platform-sdk.md](https://github.com/dotnet/docs/blob/0c13cb48cc0db577f99cab2ad3f05cdf4bb9a5b6/docs/core/testing/unit-testing-platform-sdk.md) | [Testing framework and extensions developers guide](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-sdk?branch=pr-en-us-41038) |

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@IEvangelist this is good to go

cc: @Evangelink

**Review by Evangelink — APPROVED** [review]

**Comment by IEvangelist** [comment]
We can't have an article that is a single sentence with a link elsewhere. Instead, just add a TOC entry that links to the desired location.

File: docs/core/testing/unit-testing-platform-sdk.md
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-sdk.md:1]: Delete this

File: docs/navigate/devops-testing/toc.yml [hunk 1]
```diff
@@ -163,6 +163,8 @@ items:
             href: ../../core/testing/unit-testing-platform-exit-codes.md
           - name: Integration with dotnet test
             href: ../../core/testing/unit-testing-platform-integration-dotnet-test.md
+          - name: Testing platform SDK
+            href: ../../core/testing/unit-testing-platform-sdk.md
```
Comment (IEvangelist) [inline docs/navigate/devops-testing/toc.yml:167]:
```suggestion
            href: https://github.com/microsoft/testfx/blob/main/docs/testingplatform/Index.md
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
Let's fix this a bit...

**Comment by Evangelink** [comment]
Done @IEvangelist .

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**